GDJS/Runtime/runtimescene.js

```diff
@@ -146,6 +146,26 @@ gdjs.RuntimeScene.prototype.unloadScene = function() {
 	for(var i = 0;i < gdjs.callbacksRuntimeSceneUnloaded.length;++i) {
 		gdjs.callbacksRuntimeSceneUnloaded[i](this);
 	}
+
+    this._layers = new Hashtable();
```

Comment:
I'm suprised that this was necessary 🤔
Because everytime `unloadScene` is called should be just before a scene is removed from the `SceneStack` (see https://github.com/4ian/GD/blob/master/GDJS/Runtime/scenestack.js#L55) so the associated RuntimeScene object should be removed and everything inside it should be garbage collected (unless something keeps a reference to it).

Do you know why adding this was improving the situation?

Reply:
Well, I was trying different things and looking at what sticks. The biggest issue turned out to be with audio and I found it later, but AFAIR thanks to these changes I finally started to see the memory going lower from time to time as opposed to always increasing. It could be that this isn't really necessary and would be actually collected if I waited longer, or it could be that there is some nasty reference dangling somewhere in some callback preventing it to be actually garbage collected.

So, what I can honestly tell is: I don't know. I just know that all these things together do work; some parts may be unnecessary though :)

Reply:
Thanks for the explanation :) Well let's keep this for safety, but maybe it's a good idea to add a quick comment telling that "it should not be necessary to reset these variables, but this help ensuring that all memory related to the RuntimeScene is released immediately". :)